background/background.go

```diff
@@ -2,10 +2,49 @@ package background
 
 import (
 	"github.com/bitmark-inc/autonomy-api/external/onesignal"
+	"gopkg.in/yaml.v2"
 )
 
 // Background is a struct to maintain common clients
 // and functions for all background workers
 type Background struct {
 	Onesignal *onesignal.OneSignalClient
 }
+
+// YamlDataConverter is DataConverter that uses yaml to
+// encode and decode data between workflows and activities
+type YamlDataConverter struct{}
+
+func NewYamlDataConverter() *YamlDataConverter {
+	return &YamlDataConverter{}
+}
+
+// ToData implements conversion of a list of values.
+func (c *YamlDataConverter) ToData(value ...interface{}) ([]byte, error) {
+	data := [][]byte{}
+	for _, v := range value {
+		if b, err := yaml.Marshal(v); err != nil {
+			return nil, err
+		} else {
+			data = append(data, b)
+		}
+	}
+
+	return yaml.Marshal(data)
+}
+
+// FromData implements conversion of an array of values of different types.
+// Useful for deserializing arguments of function invocations.
+func (c *YamlDataConverter) FromData(input []byte, valuePtr ...interface{}) error {
```

Comment:
There are  DataConverter: background.NewYamlDataConverter() but no use of FromData & ToData. Are they Interface functions that you pass through cadence?  What is the purpose they served

Reply:
Cadence defines the interface. I just follow its interface to add another implementation. Follow this [documentation](https://godoc.org/go.uber.org/cadence/internal#DataConverter), you will get further information.

Reply:
> Are they Interface functions that you pass through cadence? What is the purpose they served
And yes, it pass through cadence when a worker is initiated. You should find it if you search with `NewYamlDataConverter`.